backend/src/test/java/de/sambam/gamenighttracker/service/UserServiceTest.java

```diff
@@ -31,17 +35,148 @@ public void listAllGamesTest() {
                                 .build()))
                 .build()));
 
-
         UserService userService = new UserService(userDb);
 
         //WHEN
-        Optional<List<Game>> listOptional = userService.listAllGames("1");
+        List<Game> list = userService.listAllGames("1");
 
         //THEN
-        assertTrue(listOptional.equals(Optional.of(List.of(
+        assertTrue(list.equals(List.of(
                 Game.builder()
                         .name("MauMau")
-                        .build()))));
+                        .build())));
+
+    }
+
+    @Test
+    @DisplayName("listAllSessions() should return all game sessions")
+    public void listAllSessionsTest() {
+        //GIVEN
+        when(userDb.findById("1")).thenReturn(Optional.of(User.builder()
+                .id("1")
+                .userName("admin")
+                .playedGames(List.of(
+                        Game.builder()
+                                .name("MauMau")
+                                .gameSessionList(List.of(
+                                        GameSession.builder()
+                                                .id("1")
+                                                .sessionState("DONE")
+                                                .winnerPlayerId("Sanne")
+                                                .build(),
+                                        GameSession.builder()
+                                                .id("2")
+                                                .sessionState("DONE")
+                                                .winnerPlayerId("Mario")
+                                                .build()
+                                ))
+                                .build(),
+                        Game.builder()
+                                .name("Monopoly")
+                                .gameSessionList(List.of(
+                                        GameSession.builder()
+                                                .id("3")
+                                                .sessionState("DONE")
+                                                .winnerPlayerId("Hanno")
+                                                .build()
+                                ))
+                                .build()))
+                .build()));
+
+        UserService userService = new UserService(userDb);
```

Comment:
you can set up the userservice once within your test class so you do not need to do it in every test
```
private final DbClass userDb = mock(DbClass.class);
private final UserService userservice = new UserService(userDb);
```

Reply:
oh, yeah. you are right ^^